lua/possession/session.lua

```diff
@@ -362,10 +362,15 @@ function M.exists(name)
 end
 
 --- Get a list of sessions as map-like table
----@param opts? {}
+---@class possession.ListOpts
+---@field only_cwd? boolean only load sessions for cwd
+
+---@param opts? possession.ListOpts
 ---@return table<string, table> sessions {filename: session_data}
 function M.list(opts)
```

Comment:
I think this function shouldn't do the filtering of sessions. I consider it a single entry point for reading sessions and it should any avoid additional logic. The query module is to be used for filtering/etc. 

Filtering here could in theory make sense if it actually saved us from reading some files, but here we have to read the JSON data anyway, and besides this, reading a few more session files is generally pretty cheap anyway.